tests/wpunit/QueryBuilder/CRUDTest.php

```diff
@@ -214,4 +214,256 @@ public function testUpsertShouldUpdateRowInDatabase()
 		$this->assertEquals($further_updated_data['post_content'], $post->post_content);
 	}
 
+	/**
+	 * Tests if delete() can delete with both ORDER BY and LIMIT clauses.
+	 *
+	 * @return void
+	 */
+	public function testDeleteShouldWorkWithOrderByAndLimit()
+	{
+		// Insert multiple posts
+		$posts = [
+			['post_title' => 'Delete Combined A', 'post_type' => 'delete_combined_test', 'post_content' => 'Content A'],
+			['post_title' => 'Delete Combined B', 'post_type' => 'delete_combined_test', 'post_content' => 'Content B'],
+			['post_title' => 'Delete Combined C', 'post_type' => 'delete_combined_test', 'post_content' => 'Content C'],
+			['post_title' => 'Delete Combined D', 'post_type' => 'delete_combined_test', 'post_content' => 'Content D'],
+		];
+
+		$ids = [];
+		foreach ($posts as $post) {
+			DB::table('posts')->insert($post);
+			$ids[] = DB::last_insert_id();
+		}
+
+		// Delete the 2 oldest posts (lowest IDs)
+		DB::table('posts')
+			->where('post_type', 'delete_combined_test')
+			->orderBy('ID', 'ASC')
+			->limit(2)
+			->delete();
+
+		// Verify the first 2 posts were deleted
+		$post1 = DB::table('posts')
+			->where('ID', $ids[0])
+			->get();
+		$post2 = DB::table('posts')
+			->where('ID', $ids[1])
+			->get();
+
+		$this->assertNull($post1);
+		$this->assertNull($post2);
+
+		// Verify the other 2 posts still exist
+		$count = DB::table('posts')
+			->where('post_type', 'delete_combined_test')
+			->count();
+
+		$this->assertEquals(2, $count);
+	}
+
+	/**
+	 * Tests if delete() works with complex WHERE clauses using whereIn.
+	 *
+	 * @return void
+	 */
+	public function testDeleteShouldWorkWithWhereIn()
+	{
+		// Insert multiple posts
+		$posts = [
+			['post_title' => 'Delete WhereIn 1', 'post_type' => 'delete_wherein_test', 'post_content' => 'Content 1'],
+			['post_title' => 'Delete WhereIn 2', 'post_type' => 'delete_wherein_test', 'post_content' => 'Content 2'],
+			['post_title' => 'Delete WhereIn 3', 'post_type' => 'delete_wherein_test', 'post_content' => 'Content 3'],
+			['post_title' => 'Delete WhereIn 4', 'post_type' => 'delete_wherein_test', 'post_content' => 'Content 4'],
+		];
+
+		$ids = [];
+		foreach ($posts as $post) {
+			DB::table('posts')->insert($post);
+			$ids[] = DB::last_insert_id();
```

Comment:
Same deal here, I want to ensure these posts 100% exist before we test they are deleted